test/Microsoft.Azure.WebJobs.EventHubs.UnitTests/EventHubConfigurationTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Azure.EventHubs;
+using Microsoft.Azure.EventHubs.Processor;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Microsoft.Azure.WebJobs.ServiceBus;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.UnitTests
+{
+    public class EventHubConfigurationTests
+    {
+        [Fact]
+        public void Initialize_PerformsExpectedRegistrations()
+        {
+            JobHostConfiguration config = new JobHostConfiguration();
+            config.AddService<INameResolver>(new RandomNameResolver());
+
+            TestLoggerProvider loggerProvider = new TestLoggerProvider();
+            ILoggerFactory loggerFactory = new LoggerFactory();
+            loggerFactory.AddProvider(loggerProvider);
+            config.LoggerFactory = loggerFactory;
+
+            EventHubConfiguration eventHubConfiguration = new EventHubConfiguration();
+
+            IExtensionRegistry extensions = config.GetService<IExtensionRegistry>();
+            ITriggerBindingProvider[] triggerBindingProviders = extensions.GetExtensions<ITriggerBindingProvider>().ToArray();
+            Assert.Empty(triggerBindingProviders);
+            IBindingProvider[] bindingProviders = extensions.GetExtensions<IBindingProvider>().ToArray();
+            Assert.Empty(bindingProviders);
+
+            ExtensionConfigContext context = new ExtensionConfigContext
+            {
+                Config = config,
+            };
+            ((IExtensionConfigProvider)eventHubConfiguration).Initialize(context);
+
+            // ensure the EventHubTriggerAttributeBindingProvider was registered
+            triggerBindingProviders = extensions.GetExtensions<ITriggerBindingProvider>().ToArray();
+            EventHubTriggerAttributeBindingProvider triggerBindingProvider = (EventHubTriggerAttributeBindingProvider)triggerBindingProviders.Single();
+            Assert.NotNull(triggerBindingProvider);
+
+            // ensure the EventProcessorOptions ExceptionReceived event is wired up
+            var eventProcessorOptions = eventHubConfiguration.EventProcessorOptions;
+            var ex = new EventHubsException(false, "Kaboom!");
+            var ctor = typeof(ExceptionReceivedEventArgs).GetConstructors(BindingFlags.NonPublic | BindingFlags.Instance).Single();
```

Comment:
They've decided to make their event args ctor internal so you can't mock it at all :(